Index: pam/debian/local/pam-auth-update
===================================================================
--- pam.orig/debian/local/pam-auth-update
+++ pam/debian/local/pam-auth-update
@@ -570,6 +570,11 @@ sub write_profiles
 		# we have to use system because File::Copy is in
 		# perl-modules, not perl-base
 		if (-e $target && $force) {
+			my $backup = $target . '.pam-old';
+			if (-e $backup) {
+    			system('rm', '-f', $backup);
+			}
+
 			system('cp','-f',$target,$target . '.pam-old') == 0
                             or die("cp -f ${target} ${target}.pam.old failed");
 		}
