Description: Support nullok_secure as a deprecated alias for nullok
Author: Steve Langasek <vorlon@debian.org>
Last-Update: 2020-08-11

Index: pam/modules/pam_unix/support.h
===================================================================
--- pam.orig/modules/pam_unix/support.h
+++ pam/modules/pam_unix/support.h
@@ -102,8 +102,9 @@
 #define UNIX_YESCRYPT_PASS       32     /* new password hashes will use yescrypt */
 #define UNIX_NULLRESETOK         33     /* allow empty password if password reset is enforced */
 #define UNIX_OBSCURE_CHECKS      34	/* enable obscure checks on passwords */
+#define UNIX_NULLOK_SECURE       35	/* deprecated alias for nullok */
 /* -------------- */
-#define UNIX_CTRLS_              35	/* number of ctrl arguments defined */
+#define UNIX_CTRLS_              36	/* number of ctrl arguments defined */
 
 #define UNIX_DES_CRYPT(ctrl)	(off(UNIX_MD5_PASS,ctrl)&&off(UNIX_BIGCRYPT,ctrl)&&off(UNIX_SHA256_PASS,ctrl)&&off(UNIX_SHA512_PASS,ctrl)&&off(UNIX_BLOWFISH_PASS,ctrl)&&off(UNIX_GOST_YESCRYPT_PASS,ctrl)&&off(UNIX_YESCRYPT_PASS,ctrl))
 
@@ -147,6 +148,7 @@
 /* UNIX_YESCRYPT_PASS */       {"yescrypt",         _ALL_ON_^(0x6EC22000ULL),  0x40000000, 1},
 /* UNIX_NULLRESETOK */         {"nullresetok",      _ALL_ON_,                  0x80000000, 0},
 /* UNIX_OBSCURE_CHECKS */      {"obscure",          _ALL_ON_,                 0x100000000, 0},
+/* UNIX_NULLOK_SECURE */       {"nullok_secure",    _ALL_ON_^(0x200ULL),                0, 0},
 };
 
 #define UNIX_DEFAULTS  (unix_args[UNIX__NONULL].flag)
